python/equistore-operations/equistore/operations/equal.py

```diff
@@ -51,16 +51,20 @@ def equal_raise(tensor_1: TensorMap, tensor_2: TensorMap):
     :param tensor_1: first :py:class:`TensorMap`.
     :param tensor_2: second :py:class:`TensorMap`.
     """
-    try:
-        _check_same_keys(tensor_1, tensor_2, "equal")
-    except ValueError as e:
```

Comment:
Was this actually a bug that a `ValueError` was caught here and not a `NotEqualError`? Because `_check_same_keys` only raises `NotEqualError` errors. and `NotEqualError` inherits from Exception